### PR TITLE
Adding weapon stats to the output

### PR DESCRIPTION
**Pull request by philipmat**
The perks tell only part of the story - the weapon stats are very important in the ability to identify weapon classes and compare weapons of the same archetype.

This patch displays the stats available under `item.stats` for the weapon using the stats names from the API manifest.

### CODE REVIEW

File: scripts/bungie-data-helper.coffee [hunk 1]
```diff
@@ -1,13 +1,47 @@
 request = require('request')
 
 class DataHelper
+  statHashes =
```
Comment (phillipspc) [inline scripts/bungie-data-helper.coffee:4]: any reason these are hardcoded? is this not available in the definitions?

**Comment by phillipspc** [comment]
I like this addition myself, but generally speaking, we want to keep the response from the bot as concise as possible. I'm sure this info is something some users would like to see, but may not quite justify an additional line or two to every response, which adds up quickly when its being used a lot.

May do some polling of the CR slack (where this bot is used the most) before I merge this.

File: scripts/bungie-data-helper.coffee (hunk 1 above)
Comment (philipmat) [reply scripts/bungie-data-helper.coffee:4]: Good point.  
For one, they are constant (per Manifest); I was also following your example with the weapon buckets and some recommendation I read in regard to caching the DestinyStatDefinitions from the Manifest, but I see no reason, other than lookup speed, to use constants.

Let me know if you'd like me to change it to look up the `definitions.stats` instead.

**Comment by philipmat** [comment]
Pardon my ignorance - what does CR stand for? 

if you want to shop it around, here's a screenshot of before and after...

screen shot 2016-10-13 at 9 49 32 am

We use it in the Ishtar Commander slack and that's where the idea of seeing the stats came from. It seems really useful if you don't know what archetype a weapon is, or how the various perks contribute to the final stats.

File: scripts/bungie-data-helper.coffee (hunk 1 above)
Comment (phillipspc) [reply scripts/bungie-data-helper.coffee:4]: You're right, I am using some hard coded values elsewhere. This is a larger set of data, but still manageable, I say leave as is.

**Comment by phillipspc** [comment]
@philipmat CR = Crucible Radio. Its a large slack (~3000 members I think), thus the precaution for keeping the message small.

Cool to hear you guys use this in the Ishtar slack! (Amazing app btw!) I feel good about bringing this change in. I'll hopefully be able to test it out thoroughly soon. I notice in the screenshot its missing a value for "Magazine". Maybe we could have some sort of logic to handle this situation? (ie. dont show a stat if it doesnt have a value)

File: scripts/bungie-data-helper.coffee [hunk 2]
```diff
@@ -86,9 +122,13 @@ class DataHelper
...
         name = "*#{step.nodeStepName}*"
       text[column] = "" unless text[column]
-      text[column]+= name + " | "
+      text[column] += (if text[column] then ' | ' else '') + name
 
     setText node for node in nodes
-    return text
+    stats = []
+    for statName, statValue of item.stats
+        stats.push "#{statName} : #{statValue}"
```
Comment (phillipspc) [inline scripts/bungie-data-helper.coffee]: nitpicky, but could we get rid of the space before the ":" just to condense things a bit more?

File: scripts/bungie.coffee [hunk 3]
```diff
@@ -239,7 +240,7 @@ makeRequest = (bot, endpoint, callback, params) ->
   queryParams = if params then '?'+params else ''
   url = baseUrl+endpoint+trailing+queryParams
 
-  console.log("making request: #{url}")
+  # console.log("making request: #{url}")
```
Comment (phillipspc) [inline scripts/bungie.coffee]: I like having this for debugging purposes. but I can uncomment it after merging if need be

**Comment by philipmat** [comment]
Yeah, sorry about that screenshot - it was trimming out 3 characters because of the slicing to remove the pipe. I had fixed that in 6c8a666 but I had taken the screenshot before.

In testing and getting feedback, would you mind asking if people would perhaps prefer the abbreviated version of the stats instead (e.g. _RoF_, _Mag_, _Stbl_, _Rng_, etc)?

**Comment by philipmat** [comment]
Related, another idea I was considering was using the footer for the stats (top) or using the `fields` within the attachment for stats (bottom). What do you think?

screen shot 2016-10-20 at 8 45 46 am

**Comment by phillipspc** [comment]
@philipmat Nice, I like the top/footer version. Keeps the focus on the perks, but still shows the stats. Any idea how this looks on mobile? (I'd assume its fine, but just curious)

I can ask about abbreviations, but I personally prefer the full stat names. Good work on this!

**Comment by philipmat** [comment]
Looks pretty good on mobile, even if the footer is a bit small and hard to read due to being gray. 
But I like how it also conveys that the stats are secondary, a clarification of the perks, if you will. 

On Thu, Oct 20, 2016 at 9:45 AM -0500, "Peter Phillips" notifications@github.com wrote:

@philipmat Nice, I like the top/footer version. Keeps the focus on the perks, but still shows the stats. Any idea how this looks on mobile? (I'd assume its fine, but just curious)

I can ask about abbreviations, but I personally prefer the full stat names. Good work on this!

**Pull request merged**